server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/Enqueuer.java

```diff
@@ -76,6 +77,14 @@ void enQueue(Mail mail) throws MailQueue.MailQueueException {
             .block();
     }
 
+    Mono<Void> reQueue(CassandraMailQueueBrowser.CassandraMailQueueItemView item) {
+        Mail mail = item.getMail();
+        EnqueuedItem enqueuedItem = item.getEnqueuedItem();
+        return Mono.fromCallable(() -> new MailReference(enqueuedItem.getEnqueueId(), mail, enqueuedItem.getPartsId()))
```

Comment:
Building the MailReference is cheap right? We don't need to wrap it in a mono, as no serialization happens, correct?

Reply:
No, we should avoid hot publisher.